src/cli/commands/check.js

```diff
@@ -40,7 +40,7 @@ export async function run(
   }
 
   // get patterns that are installed when running `yarn install`
-  const [, rawPatterns] = await install.hydrate();
+  const [, rawPatterns] = await install.hydrate(true);
```

Comment:
I can imagine it doesn't.

What this change is mainly intended to do is to make sure that calls to `await install.linker.getFlatHoistedTree(rawPatterns);` will exclude any package that is incompatible, or any dependency of such a package.

To be able to determine if a package is incompatible, you would need to know the os and cpu requirement of packages while offline. In which case I guess it needs to be in the lock file or something?